harness/harness.sh

```
@@ -0,0 +1,134 @@
+      fi
+    done
+  fi
+  echo -e "${CYAN}━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━${NC}\n"
+}
+
+# JSON 추출 및 검증 함수
+save_state_safely() {
+  local content="$1"
+  local extracted
+  
+  extracted=$(echo "$content" | sed -n '/^```json$/,/^```$/p' | sed '1d;$d')
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)